scripts/m18_submission_checker/aida_m18_submission_checker.py

```diff
@@ -0,0 +1,381 @@
+    :param ttl_invalid_count_dict: dictionary with counts of .ttl files per directory (e.g.: {runid/invalid_dir/: 20, ...})
+    :returns: True on Success or Partial success, False otherwise
+    """
+
+    logging.info("M18 Submission Checker Report for: \n\t{0}".format(archive_file_name))
+
+    # get totals
+    count_to_be_validated = sum(ttl_valid_count_dict.values())
+    count_not_to_be_validated = sum(ttl_invalid_count_dict.values())
+
+    row_format ="{:<{col_size}} | {:<6} | {:<17}"
+
+    # print counts per each valid type directory
+    if count_to_be_validated > 0:
+        col_size = max(map(len, ttl_valid_count_dict))
+        logging.info(row_format.format("Directory", "# .ttl", "", col_size=col_size))
+        for dir, count in ttl_valid_count_dict.items():
+            logging.info(row_format.format(dir, count, "valid location", col_size=col_size))
+
+    if count_not_to_be_validated > 0:
+        col_size = max(map(len, ttl_invalid_count_dict))
+        logging.info(row_format.format("Directory", "# .ttl", "", col_size=col_size))
+        for dir, count in ttl_invalid_count_dict.items():
+            logging.warning(row_format.format(dir, count, "invalid location", col_size=col_size))
+
+
+    required_exist = None
+    if task_type == '1a' or task_type == '2':
+        #for task 1a and 2 specifically, if there are no required files we log that it is an error, else it is ok
+        for k in ttl_valid_count_dict:
+            if '/NIST/' in str(k) :
+                required_exist = (ttl_valid_count_dict[k] > 0)
+
+        if required_exist == None:
+            required_exist = False
+
+    # log archive report
+    logging.info("Total .ttl files found in submission archive: {0}".format(ttls_total_count))
+    logging.info("Total number of .ttl files to be validated: {0}".format(count_to_be_validated))
+
+    #if invalid dont exsist, dont print this line
+    if count_not_to_be_validated > 0:
+        logging.info("Total number of .ttl files that will not be validated: {0}".format(count_not_to_be_validated))
+
+    if (((task_type == '1a' or task_type == '2') and required_exist) or (count_to_be_validated > 0 and required_exist == None)) and count_not_to_be_validated == 0:
+        logging.info("SUCCESS! {0} is organized according to task {1} rules and is ready to be submitted!".format(archive_file_name, task_type))
+        return True
+    else:
+        # point user in the right direction when theres some sort of unexpected thing happening
+        logging.info("For task {0}: {1}".format(task_type, warnings_dict[task_type]))
+
+        #and also do the correct status message
+        if count_to_be_validated == 0 or not required_exist:
+            logging.error("FAILURE! {0} is not organized according to task {1} rules, or has 0 files where 1 or more is expected.".format(archive_file_name, task_type))
```

Comment:
This line is erroneously reached in the following situation:
1. Task 3 submission
2. X .ttl files in top level folder (valid)
3. Y .ttl files in a subdirectory (invalid)

This should be a PARTIAL SUCCESS because the validator will be able to validate X files but not Y files

Also:
1. Task 1b submission
2. X .ttl files in NIST/hypo1 (valid)
3. Y .ttl files in NIST/hypo2 (valid)
4. Z .ttl files in INTER-TA (invalid)

This should be a PARTIAL SUCCESS because the validator will be able to validate X+Y files but not Z files
Actually 4 above, if Z files are anywhere else other than NIST/hypo folder (e.g., in top level), then it fails when it should be a partial success.

I **think** if you switch the if and the elif then it may work right, but you should double check that.

Reply:
I just realized that my comments below describe these same issues. Sorry for the duplicate comments.

Reply:
just swapping them would mean you get a success when you expect partial success for the case of the missing NIST but existing INTER-TA

Reply:
believe i fixed this issue